summary_page.php

```diff
@@ -146,7 +146,7 @@
 <!-- LEFT COLUMN -->
 <div class="col-md-6 col-xs-12">
 
-	<?php if( 1 < count( $t_project_ids ) ) { ?>
+	<?php if( 0 < count( $t_project_ids ) ) { ?>
```

Comment:
Count can't be less than zero, so might as well remove the test (and the corresponding closing brace)

Reply:
Count maybe zero, we'd better not print a lonely table header :)

Reply:
Duh. Just ignore me :wink:

Reply:
Actually, DON'T ignore me... This code is testing whether there we are processing more than one project, so your change is introducing a regression.

The _By Project_ table should only be displayed when viewing the page for _All Projects_. With your change, the table is printed also when a single project is selected (and shows only one line).

Reply:
Re-posting @rainflood's note from https://github.com/mantisbt/mantisbt/pull/1231#discussion_r161386203 to follow a single conversation flow.

About "_By Project_"  table, we want to see the bug ratios of the only one project selected in the whole project view, also there maybe subprojects as well. Perhaps we can change the header title form "_By Project_" to "_Project_" which string name I find is "$s_email_project" in the language file, when only one project selected. 
$ grep \'Project\' lang/strings_english.txt 
$s_email_project = 'Project';
Codes in summary_page.php like this:

```php
@@ -153,7 +153,11 @@ print_summary_submenu();
                <table class="table table-hover table-bordered table-condensed table-striped">
                <thead>
                        <tr>
+                       <?php if( 1 == count( $t_project_ids ) ) { ?>
+                               <th class="width-35"><?php echo lang_get( 'email_project' ) ?></th>
+                       <?php } else { ?>
                                <th class="width-35"><?php echo lang_get( 'by_project' ) ?></th>
+                       <?php } ?>
                                <?php echo $t_summary_header ?>
                        </tr>
                </thead>
```

Reply:
The case of subprojects is already covered by the current code; the box is only hidden when showing summary for a single project. I am not opposed to having it displayed at all times, but that change was not documented as such, so I considered it as not in the PR's scope.

Quite frankly I do not see the added value of a differentiated title depending on the number of projects, it feels like needlessly complicating things. Moving forward, I would propose that

1. the change is properly documented (i.e. create an issue in our tracker to describe it)
2. reference the new issue in the message of commit 72be8c709c271d39481f3f01bc25bb6949fc582a
3. revert 2a01f01003164e81e8cb8ef8e94d9496f8890797

Reply:
> 1. the change is properly documented (i.e. create an issue in our tracker to describe it)

I just opened [#23854](https://mantisbt.org/bugs/view.php?id=23854)